src/main/java/org/alliancegenome/curation_api/controllers/crud/AGMDiseaseAnnotationCrudController.java

```diff
@@ -32,13 +32,20 @@ protected void init() {
 	}
 
 	@Override
-	public ObjectResponse<AGMDiseaseAnnotation> get(String uniqueId) {
-		SearchResponse<AGMDiseaseAnnotation> ret = findByField("uniqueId", uniqueId);
-		if (ret != null && ret.getTotalResults() == 1) {
-			return new ObjectResponse<>(ret.getResults().get(0));
-		} else {
-			return new ObjectResponse<>();
-		}
+	public ObjectResponse<AGMDiseaseAnnotation> get(String identifier) {
+		SearchResponse<AGMDiseaseAnnotation> ret = findByField("modEntityId", identifier);
+		if (ret != null && ret.getTotalResults() == 1)
+			return new ObjectResponse<AGMDiseaseAnnotation>(ret.getResults().get(0));
+		
+		ret = findByField("modInternalId", identifier);
+		if (ret != null && ret.getTotalResults() == 1)
+			return new ObjectResponse<AGMDiseaseAnnotation>(ret.getResults().get(0));
+		
+		ret = findByField("uniqueId", identifier);
+		if (ret != null && ret.getTotalResults() == 1)
+			return new ObjectResponse<AGMDiseaseAnnotation>(ret.getResults().get(0));
+				
+		return new ObjectResponse<AGMDiseaseAnnotation>();
```

Comment:
@markquintontulloch typically this kind of business logic would be in the service vs the controller. So instead of overideing the controllers get method you would do that on the service.

Reply:
@oblodgett - Done